lib/internal/stream_shared_methods.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+const { Buffer } = require('buffer');
+const errors = require('internal/errors');
+
+const errnoException = errors.errnoException;
+
+const kCreateWriteWrap = Symbol('createWriteWrap');
+const kWritevGeneric = Symbol('writevGeneric');
+const kWriteGeneric = Symbol('writeGeneric');
+
+function _handleWriteReq(req, handle, data, encoding) {
+  switch (encoding) {
+    case 'buffer':
+      return handle.writeBuffer(req, data);
+    case 'latin1':
+    case 'binary':
+      return handle.writeLatin1String(req, data);
+    case 'utf8':
+    case 'utf-8':
+      return handle.writeUtf8String(req, data);
+    case 'ascii':
+      return handle.writeAsciiString(req, data);
+    case 'ucs2':
+    case 'ucs-2':
+    case 'utf16le':
+    case 'utf-16le':
+      return handle.writeUcs2String(req, data);
+    default:
+      return handle.writeBuffer(req, Buffer.from(data, encoding));
+  }
+}
+
+const StreamSharedMethods = {
+  [kCreateWriteWrap](req, handle, oncomplete) {
+    req.handle = handle;
+    req.oncomplete = oncomplete;
+    req.async = false;
+  },
+
+  [kWritevGeneric](req, data, cb) {
```

Comment:
Just make all of these helper functions. There's no need for them to exist on the prototype. `this` can be a param of the function.